lib/game.rb

```diff
@@ -0,0 +1,159 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+   [0,1,2],
+   [3,4,5],
+   [6,7,8],
+   [0,3,6],
+   [1,4,7],
+   [2,5,8],
+   [0,4,8],
+   [2,4,6]
+  ]
+
+  # Defaults to a game with 2 human players and a new board
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    self.board = board
+    self.player_1 = player_1
+    self.player_2 = player_2
+  end
+
+  def current_player
+    x_moves = self.board.cells.select {|moves| moves == "X"}
+    o_moves = self.board.cells.select {|moves| moves == "O"}
+    x_moves.size == o_moves.size ? self.player_1 : self.player_2
+  end
+
+  def over?
+    draw? || won?
+  end
+
+
+  # Checks for winning positions occupied by X or O
+  def won?
+    WIN_COMBINATIONS.find do |win_combination|
+      # binding.pry
+      win_index_1 = win_combination[0]
+      win_index_2 = win_combination[1]
+      win_index_3 = win_combination[2]
+
+      position_1 = self.board.cells[win_index_1]
+      position_2 = self.board.cells[win_index_2]
+      position_3 = self.board.cells[win_index_3]
+
+      position_1 == position_2 && position_2 == position_3 && position_1 != " "
+    end
+  end
+
+  def draw?
+    board.full? && won? == nil
+  end
+
+  # Return winning player or nil if no one has won.
+  def winner
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.